sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py

```diff
@@ -216,11 +218,11 @@ def on_request(self, request):
                     _LOGGER.debug("    %r: %r", header, value)
                 _LOGGER.debug("Request body:")
 
-                # We don't want to log the binary data of a file upload.
-                if isinstance(http_request.body, types.GeneratorType):
-                    _LOGGER.debug("File upload")
-                else:
+                if self.logging_request_body or options.pop("logging_request_body", False):
```

Comment:
I didn't touch on_response because logging response body is handled pretty well.
For download blob, since we are doing streaming download, no actually body will be logged. For other responses logging response body is helpful, since the response body could contains raw error message or some xml elements we want to check.
Let me know if you think we should disable logging for any kinds of response body .

Reply:
We should make it symetric and possibly consider single switch for both request and response like .net and java do.
.net https://github.com/Azure/azure-sdk-for-net/search?p=1&q=IsLoggingContentEnabled&type=
java https://github.com/Azure/azure-sdk-for-java/blob/3f31d68eed6fbe11516ca3afe3955c8840a6e974/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java

Reply:
besides some may want to disable body logging due to PII being present inside.